text/0000-extern-existential-type.md

```diff
@@ -0,0 +1,217 @@
+
+One more interesting difference is the scope of where the type is transparent vs opaque: i.e. where can we see the type's definition, or only it's bounds.
+Just as in C where one gets:
+```rust
+struct Foo;
+
+// I know nothing about Foo
+
+struct Foo { int a; };
+
+// Ah now I do
+```
+when the `impl existential type` is in scope, the `existential existential type` can becomes transparent and behaves as if the declaration and definition were put together into a normal type alias.
+The definer can decide how gets to take advantage of it by making the definition public or not.
+```rust
+pub impl existential type alloc::Foo = Bar; // the big reveal
+impl existential type alloc::Foo = Bar; // the tiny reveal
+```
+There are no restrictions on the type of publicity compared to other items.
+
+Only one crate in the build plan can define the `pub extern existential type`.
+Unlike the trait system, there are no orphan restrictions that ensure crates can always be composed:
+any crate is free to define the `pub extern existential type`, as long is it isn't used with another that also does, in which case the violation will only be caught when building a crate that depends on both (or if one of the crates depends on the other).
+This is not very nice, but exactly like "lang items" and the annotations that exist today,
+so it is nothing worse.
+
+As mentioned in the introduction, code gen can be reasoned about by comparing with generic and inlining).
+We cannot generate for code for generic items until they are instantiated.
+Likewise, imagine that everything that uses an `pub extern existential type` gets an extra parameter,
+and then when the `impl pub extern existential type` is defined, we go back and eliminate that parameter by substituting the actual definition.
+Only then can we generate code.
+This is why from the root crate of compilation (the binary, static library, or other such "final" component), the dependency closure must contain an `impl existential type` for every `pub extern existential type` that's actually used.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+`pub extern existential type` can also be formally defined in reference to `existential type`.
+As explained in the guide-level explanation,
+```rust
+(pub <scope>)? extern existential type <name>: <bounds>;
+```
+creates an existential type alias that behaves just like a `use`d `existential type <name>: <bounds>` defined in another modules so it's opaque, while
+```rust
+(pub <scope>)? impl existential type <path> = <type>;
+```
+reveals the definition of the existential type alias at `path` as if it was a regular type alias.
+
+There is a post-hoc coherence rule that every used `pub extern existential type` contains exactly one `impl exisitential type` definition within the entire build-plan of crates.
```

Comment:
I had to google post-hoc; might want to explain what it means or use a different word.